src/drivers/mct/main/prep_rof_mod.F90

```diff
@@ -223,16 +325,19 @@ subroutine prep_rof_accum_avg(timer)
     character(len=*), intent(in) :: timer
     !
     ! Local Variables
-    integer :: eri, eli
+    integer :: eri, eli, eai
     character(*), parameter :: subname = '(prep_rof_accum_avg)'
     !---------------------------------------------------------------
 
     call t_drvstartf (trim(timer),barrier=mpicom_CPLID)
     do eri = 1,num_inst_rof
        eli = mod((eri-1),num_inst_lnd) + 1
        call mct_avect_avg(l2racc_lx(eli),l2racc_lx_cnt)
+       eai = mod((eri-1),num_inst_atm) + 1
+      if (rof_heat)  call mct_avect_avg(a2racc_ax(eai),a2racc_ax_cnt)
```

Comment:
I still think more changes are needed in prep_rof_accum_avg to make it completely robust. It seems like this should follow the pattern used elsewhere, where each `mct_avect_avg` call is only made if the relevant `cnt` variable is > 1: If it's 1, then it can be skipped for efficiency. If it's 0 then it's important to skip it to avoid divide by 0. Now that this is being called from multiple places, it seems like there are more ways for this to get called with one of the cnt variables being 0. I'd suggest putting conditionals around both the old `mct_avect_avg` (for l2r) and the new one (for a2r).